### PR TITLE
Migrate to sbt-typelevel

### PR DESCRIPTION
**Pull request by armanbilge**
Just to get started. Hopefully we'll be able to upstream some additional build settings and a common scalafmt config.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -1,7 +0,0 @@
-version: 2
```
Comment (cquiroz) [inline .github/dependabot.yml:1]: Is this gone because the `ci.yml` is now generated right?

File: build.sbt [hunk 2]
```diff
@@ -1,19 +1,11 @@
-
-inThisBuild(
-  (homepage := Some(url("https://github.com/gemini-hlsw/sbt-lucuma"))) +: lucumaPublishSettings
-)
+ThisBuild / tlBaseVersion := "0.5"
```
Comment (cquiroz) [inline build.sbt:1]: What's `tlBaseVersion`?

File: build.sbt [hunk 3]
```diff
@@ -1,19 +1,11 @@
...
+ThisBuild / scalaVersion := "2.12.15"
 
 lazy val sbtLucuma = (project in file("."))
-  .disablePlugins(LucumaPlugin)
   .enablePlugins(SbtPlugin)
-  .enablePlugins(AutomateHeaderPlugin)
   .settings(
     name := "sbt-lucuma",
-    addSbtPlugin("de.heikoseeberger"         % "sbt-header"             % "5.6.0"),
```
Comment (cquiroz) [inline build.sbt:12]: Are these two included in `sbt-typelevel`?

File: src/main/scala/edu/gemini/gsp/sbtplugin/LucumaPlugin.scala [hunk 4]
```diff
@@ -37,7 +36,6 @@ object LucumaPlugin extends AutoPlugin {
     lazy val lucumaPublishSettings = Seq(
       organization     := "edu.gemini",
       organizationName := "Association of Universities for Research in Astronomy, Inc. (AURA)",
-      startYear        := Some(2019),
```
Comment (cquiroz) [inline src/main/scala/edu/gemini/gsp/sbtplugin/LucumaPlugin.scala:40]: Why is this gone?

**Review by cquiroz — APPROVED** [review]
This looks good to m, Maybe @tpolecat should check it too

File: build.sbt (hunk 2 above)
Comment (armanbilge) [reply build.sbt:1]: It's part of the versioning scheme inherited from sbt-spiewak. Basically if your current version is `x.y.z` then your `tlBaseVersion` is `x.y.`. When you start working on your next version, you bump it.

https://github.com/djspiewak/sbt-spiewak/blob/main/versioning.md#compatibility-version
https://typelevel.org/sbt-typelevel/#how-do-i-introduce-breaking-changes-intended-for-my-next-version

File: .github/dependabot.yml (hunk 1 above)
Comment (armanbilge) [reply .github/dependabot.yml:1]: Yes 👍

File: build.sbt (hunk 3 above)
Comment (armanbilge) [reply build.sbt:12]: Yes.

File: src/main/scala/edu/gemini/gsp/sbtplugin/LucumaPlugin.scala (hunk 4 above)
Comment (armanbilge) [reply src/main/scala/edu/gemini/gsp/sbtplugin/LucumaPlugin.scala:40]: It's not being used AFAIK since you customize the header text.

**Comment by armanbilge** [comment]
BTW you should approve the workflow here. Also, I'm going to need workflow approvals for every commit, for my first PR to every repo 🙃

File: build.sbt (hunk 2 above)
Comment (cquiroz) [reply build.sbt:1]: I think sbt-lucuma is 0.4.4 now. Should this be `0.4`?

Comment (armanbilge) [reply build.sbt:1]: I think the changes here constitute breaking changes -> `0.5`. WDYT?

File: build.sbt (hunk 3 above)
Comment (armanbilge) [reply build.sbt:12]: Er, to clarify: sbt-tpolecat is not included in sbt-typelevel. sbt-typelevel has its own settings plugin which is effectively the same.

File: build.sbt (hunk 2 above)
Comment (cquiroz) [reply build.sbt:1]: I see

**Review by tpolecat — APPROVED** [review]
This looks good. :+1: 

What (if anything) is going to need to change in all the downstream libraries?

**Comment by armanbilge** [comment]
I'll have to figure it out on a case-by-case version, but in my experience migration typelevel libs:
1. set `tlBaseVersion`
2. clean out redundant deps/settings
3. fixing MiMa on Scala 3/JS

**Pull request merged**